### PR TITLE
Add geocode repository

### PR DESCRIPTION
**Pull request by rodrigohenriques**
Expose a repository to allow users to complete Prediction with latitude and longitude by performing a reverse geocode search on `GeocodeRepository` class passing just the place id.

### CODE REVIEW

File: library/src/main/kotlin/com/a99/rxplaces/AutocompleteEntities.kt [hunk 1]
```diff
@@ -27,12 +27,13 @@ data class MatchedSubstring(
 )
 
 data class StructuredFormatting(
-    val main_text: String,
-    val main_text_matched_substrings: List<MatchedSubstring> = listOf(),
-    val secondary_text: String = ""
+    @SerializedName("main_text") val mainText: String,
+    @SerializedName("main_text_matched_substrings")
+    val mainTextMatchedSubstrings: List<MatchedSubstring> = listOf(),
+    @SerializedName("secondary_text") val secondaryText: String = ""
```
Comment (walmyrcarvalho) [inline library/src/main/kotlin/com/a99/rxplaces/AutocompleteEntities.kt]: Maybe invert the order, so all the `@SerializedName` fields keep organized?

File: library/src/main/kotlin/com/a99/rxplaces/Exceptions.kt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.a99.rxplaces
+
+
```
Comment (walmyrcarvalho) [inline library/src/main/kotlin/com/a99/rxplaces/Exceptions.kt]: Unnecessary line.

File: library/src/main/kotlin/com/a99/rxplaces/Exceptions.kt [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+class OverQueryLimitException(message: String) : Exception(message)
+class RequestDeniedException(message: String) : Exception(message)
+class InvalidRequestException(message: String) : Exception(message)
+class UnknownErrorException(message: String) : Exception(message)
```
Comment (walmyrcarvalho) [inline library/src/main/kotlin/com/a99/rxplaces/Exceptions.kt]: Missing empty line on the end. Git merge issues, don't forget.

File: library/src/main/kotlin/com/a99/rxplaces/AutocompleteEntities.kt (hunk 1 above)
Comment (walmyrcarvalho) [inline library/src/main/kotlin/com/a99/rxplaces/AutocompleteEntities.kt]: Oh, does it follows the methods parameters on another constructor?

File: library/src/main/kotlin/com/a99/rxplaces/GoogleMapsApi.kt [hunk 4]
```diff
@@ -23,9 +23,22 @@ interface GoogleMapsApi {
       @Query("strictbounds") strictBounds: Boolean? = null
   ): Single<PlaceAutocompleteResponse>
 
+  @GET("/maps/api/geocode/json")
+  fun getReverseGeocode(
```
Comment (walmyrcarvalho) [inline library/src/main/kotlin/com/a99/rxplaces/GoogleMapsApi.kt]: So cool. <3

File: library/src/main/kotlin/com/a99/rxplaces/GoogleMapsApi.kt [hunk 5]
```diff
@@ -23,9 +23,22 @@ interface GoogleMapsApi {
...
+      @Query("key") key: String,
+      @Query("place_id") placeId: String
+  ): Single<GeocodeResponse>
+
   companion object {
     const val URL = "https://maps.googleapis.com"
 
+    const val OK = "OK"
```
Comment (walmyrcarvalho) [inline library/src/main/kotlin/com/a99/rxplaces/GoogleMapsApi.kt]: Maybe use an `Enum` with default values, instead?

File: library/src/main/kotlin/com/a99/rxplaces/AutocompleteEntities.kt (hunk 1 above)
Comment (rodrigohenriques) [inline library/src/main/kotlin/com/a99/rxplaces/AutocompleteEntities.kt]: We have only this constructor.

File: library/src/main/kotlin/com/a99/rxplaces/GoogleMapsApi.kt (hunk 5 above)
Comment (rodrigohenriques) [inline library/src/main/kotlin/com/a99/rxplaces/GoogleMapsApi.kt]: Why?

File: library/src/main/kotlin/com/a99/rxplaces/GeocodeRepositoryImpl.kt [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import com.a99.rxplaces.GoogleMapsApi.Companion.OK
+import com.a99.rxplaces.GoogleMapsApi.Companion.OVER_QUERY_LIMIT
+import com.a99.rxplaces.GoogleMapsApi.Companion.REQUEST_DENIED
+import com.a99.rxplaces.GoogleMapsApi.Companion.UNKNOWN_ERROR
+import com.a99.rxplaces.GoogleMapsApi.Companion.ZERO_RESULTS
+import rx.Observable
+
+
+internal class GeocodeRepositoryImpl (val api: GoogleMapsApi, val key: String) : GeocodeRepository {
+  override fun reverseGeocode(placeId: String): Observable<GeocodeResult> {
```
Comment (philipesteiff) [inline library/src/main/kotlin/com/a99/rxplaces/GeocodeRepositoryImpl.kt]: I suppose that observable emits just once. So try to return `Single<T>`.

Comment (philipesteiff) [inline library/src/main/kotlin/com/a99/rxplaces/GeocodeRepositoryImpl.kt]: sry for the delay :p

Comment (rodrigohenriques) [inline library/src/main/kotlin/com/a99/rxplaces/GeocodeRepositoryImpl.kt]: Yeah, sure. Once It was already merged, feel free to create a PR with this change.

**Pull request merged**